IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate.Tests/ConditionalWaitTests.cs

```diff
@@ -1,32 +1,177 @@
+        }
+
+        [Fact]
+        public async Task CheckFuncParamsForMixedInvalidTypeChecking()
+        {
+            await Assert.ThrowsAsync<ArgumentException>(
+                () => Wait.Until(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(object), typeof(Exception)));
+        }
+
+        [Fact]
+        public async Task CheckActionParamsForValidTypeChecking()
+        {
+            await Wait.Until(TestVoidDelegate, TimeSpan.FromSeconds(1), typeof(NullReferenceException));
+        }
+
+        [Fact]
+        public async Task CheckFuncParamsForValidTypeChecking()
+        {
+            await Wait.Until(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(NullReferenceException));
+        }
+
+        [Fact]
+        public async Task CheckActionsParamsForBaseTypeChecking()
+        {
+            await Wait.Until(TestVoidDelegate, TimeSpan.FromSeconds(1), typeof(Exception));
+        }
+
+        [Fact]
+        public async Task CheckFuncParamsForBaseTypeChecking()
+        {
+            await Wait.Until(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(Exception));
+        }
+
+        [Fact]
+        public async Task CheckActionParamsForMixedTypeChecking()
+        {
+            await Wait.Until(TestVoidDelegate, TimeSpan.FromSeconds(1), typeof(Exception), typeof(NullReferenceException));
+        }
+
+        [Fact]
+        public async Task CheckFuncParamsForMixedTypeChecking()
+        {
+            await Wait.Until(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(Exception), typeof(NullReferenceException));
+        }
+
+        // Might need explicit scenarios for both func and action scenarios in the below examples.
+        // Gauge based on how much Wait needs to change based on implementation.
+        // Might also need some additional testing around making sure Wait.Until actually times out when expected
+
+        [Fact]
+        public async Task CheckForUnexpectedExceptionType()
+        {
+            await Assert.ThrowsAsync<NullReferenceException>(
+                () => Wait.Until<ArgumentNullException>(
+                    () => ThrowNullRefException(), TimeSpan.FromSeconds(1)));
+        }
+
+        [Fact]
+        public async Task CheckForAggregateExceptionFromExpectedExceptionType()
+        {
+            await Assert.ThrowsAsync<AggregateException>(
+                () => Wait.Until<NullReferenceException>(
+                    () => ThrowNullRefException(), TimeSpan.FromSeconds(1)));
+        }
+
+        [Fact]
+        public async Task CheckForExpectedFailure()
+        {
+            await Assert.ThrowsAsync<AggregateException>(
+                () => Wait.Until<EqualException>(
+                    () => Equals(4, 5), TimeSpan.FromSeconds(1)));
+        }
+
+        [Fact]
+        public async Task CheckForExpectedSuccess()
+        {
+            await Wait.Until<EqualException>(() => Equals(5, 5), TimeSpan.FromSeconds(1));
```

Comment:
This test doesn't contain any expectation, so should probably be renamed to something to indicate that no exception is thrown since "Success" is pretty ambiguous.

Reply:
I thought I renamed this last night with my last commit, but apparently not... Unresolving until I can get another commit in.

Reply:
Totally confused. I pull latest and there's no test by this name. It's like the diff tool isn't picking up the change?

Reply:
Looks like you did rename it. It is only showing up in this view because you re-opened the comment thread. It no longer shows up in the branch or the files changed.